src/main/java/org/openhab/tools/analysis/checkstyle/BundleVendorCheck.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Checks if a manifest file contains the expected bundle vendor
+ *
+ * @author Martin van Wingerden
+ */
+public class BundleVendorCheck extends AbstractStaticCheck {
+    private final static String MANIFEST_EXTENSION = "MF";
+    private static final String REQUIRED_PREFIX = "Bundle-Vendor: ";
+    private static List<String> ALLOWED_VALUES = Arrays.asList("Eclipse.org/SmartHome", "openHAB");
+
+    public BundleVendorCheck() {
+        setFileExtensions(MANIFEST_EXTENSION);
+    }
+
+    @Override
+    protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+        if (isEmpty(file)) {
+            // not our task to report
+            return;
+        }
+
+        List<String> bundleVendors = lines.stream()
+                .filter(line -> line.toLowerCase().startsWith("bundle-vendor"))
+                .collect(Collectors.toList());
+
+        boolean tooMany = false;
+        if (bundleVendors.size() == 0) {
+            log(0, "\"Bundle-Vendor\" is missing");
+            return;
+        } else if (bundleVendors.size() > 1) {
+            tooMany = true;
+        }
+
+        int lineNumber = 0;
+        for (String bundleVendor : bundleVendors) {
+            lineNumber = findLineNumber(lines, bundleVendor, lineNumber);
+
+            if (tooMany) {
+                log(lineNumber, "Only one \"Bundle-Vendor\" was expected.");
+            }
+            if (!bundleVendor.startsWith(REQUIRED_PREFIX)) {
+                log(lineNumber, "Expect eg. \"Bundle-Vendor: openHAB\" got \"" + bundleVendor + "\"");
```

Comment:
I got your point, in the way it is now we will have passing tests even if we have incorrect messages. So yes, it is better to leave the tests with hardcoded messages,